src/utils.js

```diff
@@ -3,6 +3,18 @@ export function extract(obj, path, dflt) {
   return path.split('.').reduce((obj, key) => (obj || $n)[key], obj) || dflt;
 }
 
+export function isEmpty(value) {
+  let length = 0;
+
+  if (typeof value === 'object') {
+    length = value instanceof Set ? value.size : Object.keys(value).length;
```

Comment:
I was having trouble with `Set` also being `typeof object` -turns out you were right earlier that there is more than one kind of object :)
I made this to simplify checking against our different query types which can be sets, objects or arrays, using the appropriate length method for each.

Reply:
Rather than nesting a ternary inside the `if`, why not make it every case explicit:

```
if (Set.prototype.isPrototypeOf(value)) {
   return !value.size;
} else if (Array.isArray(value)) {
  return !value.length;
} else if (typeof value === 'object') {
  return !Object.keys(value).length;
} else {
  return !value;
}
```

Reply:
ah much better.